sdk/core/core/src/az_span.c

```diff
@@ -153,6 +154,30 @@ AZ_NODISCARD az_result az_span_atou32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_atod(az_span span, az_span* conversion_buffer, double* out_number)
```

Comment:
This conversion buffer and byte copy requirement is not ideal. Why do we need this function? Can we just kill it (for now).